### ISSUE TITLE
[Bug]: IDs and aria-... changing on subsequent snapshot testing 

### ISSUE THREAD

**Issue by Cam-Bloom** [microsoft/fluentui#28237]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700
    Memory: 4.09 GB / 15.70 GB
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (114.0.1823.43)
    Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

None

### Reproduction

Not able to reproduce in codesandbox 

## Setup 
I have a Fluent UI Card component structure of which can be found below, I am testing this with react testing library and jest snapshot testing, test can be found below, test passes locally roughly 19/20 times, other time either the id, aria-describedby or aria-labelledby changes (Diff logs can be found below), afterward test passes again. Test also fails always in CI CD for the same reason.

Specs
Jest 29.5.0
@fluentui/react-components 9.20.0
@testing-library/react 13.4.0

## Actual Behavior
Occasionally snapshot fails due to different props.
![image](https://github.com/microsoft/fluentui/assets/117858656/3d070711-aabe-40d2-8a5a-436b82f0837a)

## Expected Behavior
Snapshot to match every time unless code changes

## Component Structure 
```
  <div>
      <Card>
        <CardHeader
          image={
            <Tooltip
              relationship="description"
              positioning="below"
              hideDelay={0}
              content={"adfadsfdsa"}
              withArrow={true}
            >
              <Button appearance="transparent" />
            </Tooltip>
          }
          header={
            <div>
              <Subtitle2>{"DocIndex.Name"}</Subtitle2>
              <Tooltip
                relationship="label"
                positioning="below"
                withArrow={true}
                content={""}
              >
                <Badge appearance="tint" size="small">
                  fdsafdsaklj
                </Badge>
              </Tooltip>

              <Badge appearance="tint" size="small">
                'fjdsalfasdkl'
              </Badge>
            </div>
          }
          description={
            <div>
              <Caption1>xxxx</Caption1>
              <Caption1>xxxx </Caption1>
            </div>
          }
          action={
            <div>
              <Menu>
                <MenuTrigger>
                  <Button appearance="transparent" />
                </MenuTrigger>

                <MenuPopover>
                  <MenuList>
                    <MenuItem>"document-search.view-full-details"</MenuItem>
                    <MenuItem>"document-search.view-details-split"</MenuItem>
                    <MenuItem>"document-search.view-stored-file"</MenuItem>
                  </MenuList>
                </MenuPopover>
              </Menu>
            </div>
          }
        />
      </Card>
    </div>
  );
}
```

## Test
    test('Should match snapshot of card', async () => {
        const { container } = customRender(
            <DocIndexCard
               ...props
            />
        );
        await screen.findByText('Revision 1.0');
        expect(container).toMatchSnapshot();
    });

### Logs

```shell
expect(received).toMatchSnapshot()

    Snapshot name: `Document Index Card Should match snapshot of card 1`

    - Snapshot  - 12
    + Received  + 10

    @@ -1,42 +1,41 @@
      <div>
        <div
    -     class="fui-FluentProvider fui-FluentProviderr1cj ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    +     class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
          dir="ltr"
        >
          <div
            class=

                {
    -             -moz-osx-font-smoothing: grayscale;
    -             -webkit-font-smoothing: antialiased;
    -             background: #ffffff;
    -             color: #323130;
    -             font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
    -             font-size: 14px;
    -             font-weight: 400;
    +             -moz-osx-font-smoothing: grayscale;
    +             -webkit-font-smoothing: antialiased;
    +             background: #ffffff;
    +             color: #323130;
    +             font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
    +             font-size: 14px;
    +             font-weight: 400;
                }
          >
            <div>
              <div
                class="fui-Card ___15pjoyd_ui0c590 f1p9o1ba f1sil6mw fifeqxg f899z7z f4h3tyx f18ur2pz fikn0iw fw5db7e fdxej3c f1uw59to fxsr4vj fcvsdzp f22iagw f10pi13n f1ewtqcl f19n0e5 f1mdlcz9 frwkxtg f1n6gb5g f15yvnhg fo72kxq f13zj6fq f1nwj1ja f8rth92 flthirb f1lh990p ftkbnf5 f6czdpx f13hvwk3 f1i1u9k0 f1en4csx f1qnomq5 f2fl922 f1anhtl f1n2zcl3 f16v3d5c f1su8t2g ftqa4ok f2hkw1w f8hki3x f1d2448m f1bjia2o ffh67wi f15bsgw9 f14e48fq f18yb2kv fd6o370 fh1cnn4 fy7oxxb fpukqih f184ne2d f99gebs f13b0oaq f1jvq617 f8t2bz6 f11unbnk fbd201q f12nqxso f1uguk4w f1dvezut fd0oaoj fjvm52t f1cwg4i8 f3l4wcz f6j2biq fdzzmfx fduh8kh f1vx9l62 fobhde4 fx5r7kn f19chtn8 fuvs6re fy4glsf frsmuga fuldkky fxugw4r f1whvlc6 f16gxe2i fpgykix f1osi826 fzybk4o f1k6fduh f1knas48 f1m145df fb40n2d fpfvv3l f1oamsm6 f1il21bs fnn5dk0 fey3rwa f5jhx11 fm7n0jy fff9uym f1rgks96 f3052tw f1lmfglv fibjyge f18zxyen f199hnxi"
                data-testid="docIndexCard"
                role="group"
              >
                <input
    -             aria-labelledby="fui-CardHeader__headerr1cl"
                  class="fui-Card__checkbox ___rou8bv0_6d15r90 f1p9o1ba f1sil6mw frkrog8 f1mpe4l3 f1euv43f fmf1zke f1wch0ki fz5stix"
                  type="checkbox"
                />
                <div
                  class="fui-CardHeader ___uu8k3y0_1igg196 fkufhic f13qh94s f8a668j f4h39za f122n59"
                >
                  <div
                    class="fui-CardHeader__image ___c1l6a60_1j4wpwx ftuwxu6 fql5097 fwpfdsa fldnz9j"
                  >
                    <button
    -                 aria-describedby="tooltip-r1cm"
    +                 aria-describedby="tooltip-r3"
                      class="fui-Button ri7kuyr ___91x5cg0_tj6h1h0 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1sbtcvk fwiuce9 fdghr9 f15vdbe4 fwbmr0d f44c6la"
                      type="button"
                    >
                      <span
                        class="fui-Button__icon rywnvv2"
    @@ -64,11 +63,10 @@
                      hidden=""
                    />
                  </div>
                  <div
                    class="fui-CardHeader__header ___pi3n290_15s0baj fd46tj4 f16hsg94"
    -               id="fui-CardHeader__headerr1cl"
                  >
                    <div>
                      <span
                        class="fui-Subtitle2 fui-Text ___ut231u0_1ul1zvu fk6fouc fod5ikn faaz57k fl43uef fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
                      >
    @@ -155,11 +153,11 @@
                    >
                      <button
                        aria-haspopup="menu"
                        aria-label="More options"
                        class="fui-Button ri7kuyr ___91x5cg0_tj6h1h0 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1sbtcvk fwiuce9 fdghr9 f15vdbe4 fwbmr0d f44c6la"
    -                   id="menur1co"
    +                   id="menur5"
                        role="button"
                        tabindex="0"
                        type="button"
                      >
                        <span

      129 |         );
      130 |         await screen.findByText('Revision 1.0');
    > 131 |         expect(container).toMatchSnapshot();
          |                           ^
      132 |     });
      133 | });
```

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by marcosmoura** [comment]
@Cam-Bloom Could you provide a Code Sandbox example with the code that causes this issue?

**Comment by Cam-Bloom** [comment]
@marcosmoura
Morning, here is a version of the code that causes the issue, there has been some small changes due to company policy but overall structure is the same, test wont run in browser due to jest-mock but should work in a local environment. If you require anything else please let me know.

https://codesandbox.io/s/determined-babbage-37rq5s?file=/src/IndexCard.tsx

**Comment by marcosmoura** [comment]
This is an intended behaviour.

TL:DR:
You just need to add an id parameter to the `<CardHeader />` component, so we don't need to generate one internally. We do this for A11y purposes.

Full explanation:
When using the `onSelectionChange` the card becomes "selectable". Selectable cards have an internal checkbox that is invisible, but it is there to represent the selection state, and for accessibility purposes (especially narration) we need to specify aria attributes for this checkbox. As you are not providing any accessibility attributes, we need to guess the best elements to use for narration and create IDs for them so we can reference them using aria attributes in this internal checkbox. The elements we use for that are the CardHeader and CardPreview. That's why the IDs can be different from time to time.
